### PR TITLE
Add path-prefix flag, replace hard code `skydns` in etcd prefix path,defaults to `skydns`

### PR DESCRIPTION
**Pull request by weekface**

### CODE REVIEW

File: msg/service.go [hunk 1]
```diff
@@ -12,6 +13,10 @@ import (
 	"github.com/miekg/dns"
 )
 
+var PathPrefix string
+
+const PATH_PREFIX string = "skydns"
```
Comment (miekg) [inline msg/service.go]: should this default be `/skydns/` in this setup?

File: msg/service_test.go [hunk 2]
```diff
@@ -6,6 +6,16 @@ package msg
 
 import "testing"
 
+func TestPath(t *testing.T) {
+	PathPrefix = "skydns"
```
Comment (miekg) [inline msg/service_test.go]: Why not test with something else than "skydns"

File: msg/service.go [hunk 3]
```diff
@@ -12,6 +13,10 @@ import (
 	"github.com/miekg/dns"
 )
 
+var PathPrefix string
```
Comment (miekg) [inline msg/service.go]: Wouldn't it be easier to let user just override this var?

```
var PathPrefix string = "skydns"
```

With a proper comment?

File: msg/service.go [hunk 4]
```diff
@@ -118,10 +123,10 @@ func PathWithWildcard(s string) (string, bool) {
 	}
 	for i, k := range l {
 		if k == "*" || k == "any" {
-			return path.Join(append([]string{"/skydns/"}, l[:i]...)...), true
+			return path.Join(append([]string{Pathprefix()}, l[:i]...)...), true
 		}
 	}
-	return path.Join(append([]string{"/skydns/"}, l...)...), false
+	return path.Join(append([]string{Pathprefix()}, l...)...), false
```
Comment (miekg) [inline msg/service.go]: this could then just become "/" + PathPrefix + "/"

File: msg/service.go (hunk 1 above)
Comment (weekface) [inline msg/service.go]: Yes: https://github.com/weekface/skydns/commit/2afffe2c896223e5a7f792bd11a6a35efab6225a#diff-7ddfb3e035b42cd70649cc33393fe32cR72

**Comment by weekface** [comment]
@miekg fixed.

File: msg/service.go [hunk 5]
```diff
@@ -12,6 +12,8 @@ import (
 	"github.com/miekg/dns"
 )
 
+var PathPrefix string = "skydns"
```
Comment (miekg) [inline msg/service.go]: All look good. But can you add some docs to this export var? I.e. along the lines of the docs you added to skydns it self.

Comment (weekface) [inline msg/service.go]: Ok

**Comment by weekface** [comment]
@miekg docs added, pls check it.

**Comment by miekg** [comment]
Thanks. LGTM.

**Comment by miekg** [comment]
Just made small change to the docs.

Thanks!

**Comment by weekface** [comment]
Good.

**Pull request merged**